shortcode.php

```diff
@@ -330,3 +330,110 @@ function geoip_detect_shortcode_user_info($attr) {
+    $atts_array = shortcode_atts(array(
+        'timezone' => null,
+        'continent' => null,
+        'country' => null,
+        'region' => null,
+        'state' => null,
+        'city' => null,
+        'and-not' => null),
+        $atts);
+
+    $info = geoip_detect2_get_info_from_current_ip();
+    $criteria_test_flag = true;       // If set to false, nothing will display
+    
+    /* Attribute Conditions */
+    // Timezone
+    if ($atts_array['timezone'] != null) {
+        if ($info->location->timeZone && $atts_array['timezone'] != $info->location->timeZone) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Continent
+    if ($atts_array['continent'] != null) {
+        if ($info->continent->code && $atts_array['continent'] != $info->continent->code) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Country
+    if ($atts_array['country'] != null) {
+        if ($info->country->name && $atts_array['country'] != $info->country->name
+            && $info->country->isoCode && $atts_array['country'] != $info->country->isoCode) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Region/State
+    if ($atts_array['region'] != null) {
+        if ($info->mostSpecificSubdivision->name && $atts_array['region'] != $info->mostSpecificSubdivision->name
+        && $info->mostSpecificSubdivision->isoCode && $atts_array['region'] != $info->mostSpecificSubdivision->isoCode) {
+            $criteria_test_flag = false;
+        }
+    }
+    if ($atts_array['state'] != null) {
+        if ($info->mostSpecificSubdivision->name && $atts_array['state'] != $info->mostSpecificSubdivision->name
+            && $info->mostSpecificSubdivision->isoCode && $atts_array['state'] != $info->mostSpecificSubdivision->isoCode) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // City
+    if ($atts_array['city'] != null) {
+        if ($info->city->name && $atts_array['city'] != $info->city->name) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // And Not (Exclusion)
+    if ($atts_array['and-not'] != null) {
+        if ($info->location->timeZone && $atts_array['and-not'] == $info->location->timeZone
+        || $info->continent->code && $atts_array['and-not'] == $info->continent->code
+        || $info->country->name && $atts_array['and-not'] == $info->country->name
+        || $info->country->isoCode && $atts_array['and-not'] == $info->country->isoCode
+        || $info->mostSpecificSubdivision->name && $atts_array['and-not'] == $info->mostSpecificSubdivision->name
+        || $info->mostSpecificSubdivision->isoCode && $atts_array['and-not'] == $info->mostSpecificSubdivision->isoCode
+        || $info->city->name && $atts_array['and-not'] == $info->city->name) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // All Criteria Passed?
+    if ($criteria_test_flag == true) {
+        return do_shortcode($content);
+    }
```

Comment:
I'm not sure if I understand what you mean. Would this do?
```
    if ($criteria_test_flag == true) {
        if ($content != null)
            return do_shortcode($content);
    }
```

Reply:
Will do.

Reply:
Oh sorry, that's not what I meant. 

In PHP, when there is no return statement, it will return null implicitly - which is different to an empty string.
```php
     if ($criteria_test_flag == true) {
        return do_shortcode($content);
     }
     return '';
```